prov/efa/src/rdm/efa_rdm_peer.c

```diff
@@ -190,6 +233,52 @@ int efa_rdm_peer_reorder_msg(struct efa_rdm_peer *peer, struct efa_rdm_ep *ep,
 	return 1;
 }
 
+/**
+ * @brief After processing recvwin_size of pkt entries, iterate over the
+ * overflow_pke_list and move the pkt entry to receive window if it fits.
+ *
+ * @param[in,out]	peer 		peer
+ * 
+ */
+void efa_rdm_peer_move_overflow_pke_to_recvwin(struct efa_rdm_peer *peer)
+{
+	struct efa_rdm_peer_overflow_pke_list_entry *overflow_pke_list_entry;
+	struct efa_rdm_pke *overflow_pkt_entry;
+	struct dlist_entry *tmp;
+	uint32_t msg_id;
+	int ret;
+
+	if (dlist_empty(&peer->overflow_pke_list)) {
+		return;
+	}
+
+	dlist_foreach_container_safe (
+		&peer->overflow_pke_list,
+		struct efa_rdm_peer_overflow_pke_list_entry,
+		overflow_pke_list_entry, entry, tmp) {
+		overflow_pkt_entry = overflow_pke_list_entry->pkt_entry;
+		msg_id = efa_rdm_pke_get_rtm_msg_id(overflow_pkt_entry);
+		if (ofi_recvwin_id_valid((&peer->robuf), msg_id)) {
+			ret = efa_rdm_peer_recvwin_queue_or_append_pke(
+				overflow_pkt_entry, msg_id, (&peer->robuf));
+			if (OFI_UNLIKELY(ret == -FI_ENOMEM)) {
```

Comment:
I would just check `ret < 0`. I think `efa_rdm_peer_recvwin_queue_or_append_pke` just has a rule that it will return negative integer when error happens.

Reply:
Do we still efa_base_ep_write_eq_error FI_EFA_ERR_OOM?

Reply:
I see, I see the current caller of this function also check OOM explicitly. Then keep the current change as is.